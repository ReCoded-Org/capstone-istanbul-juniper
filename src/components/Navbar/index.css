.navbar {
  background: #3b8c79;
  height: 15vh;
}

.navbar__topHalf {
  margin: 0;
  padding: 0;
  background-color: #3b8c79;
  font-family: "Salsa", cursive;
  position: relative;
  top: 0;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  color: white;
  height: 50%;
}

.navbar__title {
  color: white;
  font-size: 2em;
}

.navbar__topHalf ul li a {
  color: white;
}

.navbar__topHalf ul {
  list-style: none;
  display: flex;
  flex-direction: row;
  width: 25%;
  justify-content: space-between;
  align-items: stretch;
  font-size: 1.3em;
}

.navbar__topHalf ul li:hover,
.navbar__title:hover {
  cursor: pointer;
}

.navbar__helperDiv {
  height: 50%;
}

.navbar__bottomHalf {
  position: relative;
  background: #b4d0c4;
  border-radius: 50% 50% 0 0;
  height: 50%;
}

.navbar__logo {
  z-index: 1;
  position: absolute;
  display: inline-block;
  padding: 0px;
  height: 6vh;
  width: auto;
}

.navbar__logo-link{
  position: absolute;
  left: 47.9%;
  top: 5vh;
  width: 3%;
  height: auto;
}

.navbar__logo:hover {
  cursor: pointer;
}

.navbar__dropdown {
  position: absolute;
  left: 5vw;
  margin: 0;
  background-color: transparent;
  width: 1.2em;
  height: 1.2em;
}

/* antdesign class */
.ant-dropdown {
  font-family: "Salsa";
}

/* antdesign class */
.anticon-menu .svg:hover {
  cursor: pointer;
}

/* antdesign class */
.anticon-menu .svg:focus {
  box-shadow: #b4d0c4;
}

.navbar__dropdown-btn {
  width: 1.2vh;
  height: 1.2vh;
  background: transparent;
}

.navbar__leftSide {
  display: flex;
  justify-content: space-between;
}

.navbar__rightSide {
  display: flex;
  justify-content: stretch;
}

.navbar__languageContainer-btn {
  background-color: #3b8c79;
  border-width: 1px;
  border-color: #b4d0c4;
  color: white;
  width: 30%;
  height: auto;
}


.navbar__languageContainer-btn:hover {
  cursor: pointer;
  background-color: #b4d0c4;
}

.navbar__languageContainer {
  display: flex;
  justify-content: space-around;
  position: absolute;
  right: 1vw;
  width: 10%;
  height: 27px;
  top: 70%;
}

@media (max-width: 1024px) {
  .navbar__rightSide li a {
    display: none;
  }
  .navbar__leftSide li a {
    display: none;
  }

}

@media (min-width: 1024.1px) {
  .navbar__dropdown {
    display: none;
  }
}

@media (max-width: 425px) {
  .navbar__languageContainer{
    width: 23%;
    top: 3vh;
  }
}