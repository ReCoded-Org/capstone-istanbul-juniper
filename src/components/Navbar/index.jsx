import React from "react";
import "./index.css";
import { Link, BrowserRouter as Router } from "react-router-dom";
import { Dropdown, Menu } from "antd";
import { MenuOutlined } from "@ant-design/icons";
import { useTranslation } from "react-i18next";
import i18n from "i18next";

const logoImg = require("./../../images/logo.svg");

const Navbar = () => {
  const { t } = useTranslation();

  const menu = (
    <Menu>
      <Menu.Item>
        <Link to="/">{t("navbar.home")}</Link>
      </Menu.Item>
      <Menu.Item>
        <Link to="/about">{t("navbar.about")}</Link>
      </Menu.Item>
      <Menu.Item>
        <Link to="/games">{t("navbar.games")}</Link>
      </Menu.Item>
      <Menu.Item>
        <Link to="/resources">{t("navbar.resources")}</Link>
      </Menu.Item>
      <Menu.Item>
        <Link to="/contact">{t("navbar.contact")}</Link>
      </Menu.Item>
      <Menu.Item>
        <Link to="/login">{t("navbar.login")}</Link>
      </Menu.Item>
    </Menu>
  );

  const handleLanguageChange = (lang) => {
    i18n.changeLanguage(lang);
  };

  return (
    <Router>
      <div className="navbar">
        <div className="navbar__topHalf">
          {/*Dropdown replaces the navbar in small screens */}
          <Dropdown
            className="navbar__dropdown"
            overlay={menu}
            placement="bottomleft"
          >
            <button className="navbar__dropdown-btn">
              <MenuOutlined />
            </button>
          </Dropdown>

          <ul className="navbar__leftSide">
            <li>
              <Link to="/">{t("navbar.home")}</Link>
            </li>
            <li>
              <Link to="/about">{t("navbar.about")}</Link>
            </li>
            <li>
              <Link to="/games">{t("navbar.games")}</Link>
            </li>
          </ul>
          <Link to="/">
            <h1 className="navbar__title">Juniper</h1>
          </Link>
          <ul className="navbar__rightSide">
            <li>
              <Link to="/resources">{t("navbar.resources")}</Link>
            </li>
            <li>
              <Link to="/contact">{t("navbar.contact")}</Link>
            </li>
            <li>
              <Link to="/login">{t("navbar.login")}</Link>
            </li>
          </ul>
          <Link className="navbar__logo-link" to={"/"}>
            <img
              className="navbar__logo"
              src={logoImg}
              alt="juniper-logo"
            />
          </Link>
          <div className="navbar__languageContainer">
              <button
                className="navbar__languageContainer-btn"
                onClick={(e) => handleLanguageChange("en")}
              >
               <p>En</p>
              </button>
              <button
                className="navbar__languageContainer-btn"
                onClick={(e) => handleLanguageChange("tr")}
              >
               <p>Tr</p>
              </button>
              <button
                className="navbar__languageContainer-btn"
                onClick={(e) => handleLanguageChange("ar")}
              >
                <p>Ar</p>
              </button>
            </div>
        </div>

        <div className="navbar__bottomHalf">
          <div className="navbar__helperDiv"></div>
          <div>
          </div>
        </div>
      </div>
    </Router>
  );
};

export default Navbar;
