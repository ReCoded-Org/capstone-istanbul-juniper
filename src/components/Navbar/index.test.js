import React from "react";
import Navbar from "./index.jsx";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

describe("Navbar Component", () => {
  it("matches the tree", () => {
    const tree = renderer
      .create(
        <Router>
          <Navbar />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
