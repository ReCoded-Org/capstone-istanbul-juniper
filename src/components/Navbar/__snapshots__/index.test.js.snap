// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches the tree 1`] = `
<div
  className="navbar"
>
  <div
    className="navbar__topHalf"
  >
    <button
      className="ant-dropdown-trigger navbar__dropdown-btn navbar__dropdown"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <span
        aria-label="menu"
        className="anticon anticon-menu"
        role="img"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="menu"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
    </button>
    <ul
      className="navbar__leftSide"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          navbar.home
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          navbar.about
        </a>
      </li>
      <li>
        <a
          href="/games"
          onClick={[Function]}
        >
          navbar.games
        </a>
      </li>
    </ul>
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="navbar__title"
      >
        Juniper
      </h1>
    </a>
    <ul
      className="navbar__rightSide"
    >
      <li>
        <a
          href="/resources"
          onClick={[Function]}
        >
          navbar.resources
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          navbar.contact
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          navbar.login
        </a>
      </li>
    </ul>
    <a
      className="navbar__logo-link"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="juniper-logo"
        className="navbar__logo"
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "logo.svg",
          }
        }
      />
    </a>
    <div
      className="navbar__languageContainer"
    >
      <button
        className="navbar__languageContainer-btn"
        onClick={[Function]}
      >
        <p>
          En
        </p>
      </button>
      <button
        className="navbar__languageContainer-btn"
        onClick={[Function]}
      >
        <p>
          Tr
        </p>
      </button>
      <button
        className="navbar__languageContainer-btn"
        onClick={[Function]}
      >
        <p>
          Ar
        </p>
      </button>
    </div>
  </div>
  <div
    className="navbar__bottomHalf"
  >
    <div
      className="navbar__helperDiv"
    />
    <div />
  </div>
</div>
`;
